drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -71,6 +57,24 @@ local function device_init(driver, device)
   device:subscribe()
 end
 
+local function device_added(driver, device)
+  local switch_eps = device:get_endpoints(clusters.OnOff.ID)
+  table.sort(switch_eps)
+
+  local endpoint_map = {}
+  local current_component_number = 1
+  for _, ep in ipairs(switch_eps) do
+    if current_component_number == 1 then
+      endpoint_map[ep] = "main"
+    else
+      endpoint_map[ep] = string.format("switch%d", current_component_number)
+    end
+    current_component_number = current_component_number + 1
+  end
+
+  device:set_field(ENDPOINT_TO_COMPONENT_MAP, endpoint_map, {persist = true})
+end
```

Comment:
Is there any concern with race conditions on creating and populating the map within `device_added` compared to `do_configure`? Or is the general flow that `device_added` will be called before any other operation can take place for the driver?

Reply:
It looks like the lifecycle handlers do invoke the added first. @ctowns please correct if this is not correct.

Reply:
Good question, Tejus. Eugen is correct, device_add is queued and handled before the doConfigure event in this case.